src/stdlib_string_type.f90

```diff
@@ -0,0 +1,1102 @@
+! SPDX-Identifier: MIT
+
+!> Implementation of a string type to hold an arbitrary sequence of characters.
+!>
+!> This module provides string type compatible with all Fortran instrinsic character
+!> procedures as well as overloaded operators for working with character variables.
+!>
+!> A string type can be easily constructed by creating a new instance from a
+!> character variable or literal by invoking its constructor or by assigning it
+!> to a string type. Generally, the string type behaves similar to a deferred
+!> length character in most regards but adds memory access safety.
+!>
+!> The specification of this module is available [here](../page/specs/stdlib_string_type.html).
+module stdlib_string_type
+    implicit none
+    private
+
+    public :: string_type
+    public :: len, len_trim, trim, index, scan, verify, repeat, adjustr, adjustl
+    public :: lgt, lge, llt, lle, char, ichar, iachar
+    public :: assignment(=)
+    public :: operator(.gt.), operator(.ge.), operator(.lt.), operator(.le.)
+    public :: operator(.eq.), operator(.ne.), operator(//)
+    public :: write(formatted), write(unformatted)
+    public :: read(formatted), read(unformatted)
+
+
+    integer, parameter :: long = selected_int_kind(18)
+
+
+    !> String type holding an arbitrary sequence of characters.
+    type :: string_type
+        sequence
```

Comment:
```suggestion
    type :: string_type
        ! Use the sequence statement below as a hack to prevent extending this type.
        ! It is not used for storage association.
￼        sequence
```